### PR TITLE
Contributions for secwiki.cloud: Adding a Cognito page

### PR DESCRIPTION
**Pull request by Snoarlax**
Fixed duplicate list element in API_Gateway.md

Fixed Typo in CloudFront.md

Added Cognito page

Added name to contributors

### CODE REVIEW

**Comment by x-delfino** [comment]
well I need to fix the linting again - looks like it was always fail with checkin, but the rule runs are still accurate and some of those have failed and can be checked in 'details' in the 'Lint code base' task

**Comment by x-delfino** [comment]
ta - looking good on formatting now. have assigned to @matthewkeogh50 as I believe he's agreed to qa

**Comment by matthewkeogh50** [comment]
A general point for the Cognito page - it's worth cutting back on the bolding. As a general point, unless you are making an explicit point, you don't need to bold. Currently, it has lost its value because there is too much bolded stuff.

**Comment by Snoarlax** [comment]
Sure, I can remove the bolding on keywords?

File: docs/aws/services/Cognito.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+- Cognito allows for email addresses to be changed through the CLI. If this is misconfigured to reflect changes before they are confirmed via an email/SMS code, a downstream service could potentially use the unconfirmed address when authenticating the user. This would result in an account takeover vector. See [Flickr account takeover](https://security.lauritz-holtmann.de/advisories/flickr-account-takeover/) for more detail.
+
+**Firstly, when configuring the user pool ensure Keep original attribute value active when an update is pending is enabled. This would stop changes in the email address from being reflected until it is verified. Secondly, read/ write access control can be applied on user attributes. Limit read/write access to attributes which are used in downstream services, especially for authentication and authorisation. Lastly, avoid using the 'email' attribute when authenticating a user. Instead, use the 'sub' attribute, which is guaranteed to be unique for each user.**
```
Comment (matthewkeogh50) [inline docs/aws/services/Cognito.md:78]: Maybe just bullet point these to make it a litter easier to go through.

File: docs/aws/services/Cognito.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+aws cognito-identity get-id --identity-pool-id <identity pool id> --login <ISS>=<Identity token>
+```
+
+```bash
+cognito-identity get-credentials-for-identity --identity-id <identity id> --login <ISS>=<Identity token>
+```
+
+*\<ISS\> is equivalent to the issuer of the token, which can be found inside the payload of the ID token JWT*
+
+**When configuring an identity pool, apply the principle of least privilege. Disable the unauthenticated role if possible, and apply the minimum privileges required for the authenticated role using IAM policies.**
```
Comment (matthewkeogh50) [inline docs/aws/services/Cognito.md:112]: Any example policies or AWS docs we can ref here? Or maybe an example IAM policy for a simple webapp that is authing via Cognito?

**Review by matthewkeogh50 — CHANGES_REQUESTED** [review]

File: docs/aws/services/Cognito.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+- User pools
+- Identity pools
+
+A **user pool** is a directory of users . It integrates with other SSO providers such as Google, Amazon and Apple. This allows customers to create a user account in the **user pool** by simply authenticating with their chosen SSO provider. It uses **OpenID Connect** or **SAML** to delegate authentication to third party identity providers. Additionally, it can be used to store custom attributes about users. Users can be assigned read/ write permissions to these attributes when the **user pool** is configured.
+
+An **identity pool** allows users to access the accounts AWS resources, by provisioning a specific **IAM role** which a person who is authenticated with a **user pool** can assume. It can also allow unauthenticated users to assume a different **IAM role** if configured to do so.
+
+Listed below are the common terms, grant flows and types of client required to understand Cognito
+
+| Name                | Type       | Description                                                  |
```
[the remark this answers is not in the document]
Comment (Snoarlax) [reply docs/aws/services/Cognito.md:16]: Good point, thanks!

Open Authorisation is a protocol identity servers use to implement fine grained access control on some service to a resource server

Re: JWT, want me to add another row describing it or do it inside the description of ID token? (access token doesn't reference JWTs)

File: docs/aws/services/Cognito.md (hunk 2 above)
Comment (Snoarlax) [reply docs/aws/services/Cognito.md:112]: Thing is its so context dependent, I'm not really sure what to give as an example. At this point, its just saying to configure IAM roles properly, which is its whole section on its own. Perhaps I should link back to the IAM page on the wiki or something?

File: docs/aws/services/Cognito.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+| Implicit grant      | Grant flow | The user directly receives an **access token** after authenticating with the **identity server**. |
+| Confidential client | Client     | The client is considered confidential, meaning **client secrets** can be stored on it. An example would be a backend server. |
+| Public client       | Client     | The client is not confidential, meaning **client secrets** cannot be stored on it. An example would be a web application. |
+
+## Assessment Notes
+
+The methodology for assessing the security of Cognito differs when attacking a **user pool** or an **identity pool**. **User pools** are easy to misconfigure, potentially allowing for access tokens to be retrieved by the user to run AWS CLI commands with. It is also important to consider how downstream applications handle **claims**, **assertions** and **user attributes**. These can sometimes be directly changed by users with an access token through the AWS CLI, which can potentially lead to privilege escalation or account takeover.
```
[the remark this answers is not in the document]
Comment (Snoarlax) [reply docs/aws/services/Cognito.md]: Updated to be a bit more specific

**Pull request merged**